ibis/backends/sql/compilers/base.py

```diff
@@ -469,6 +460,19 @@ def impl(self, _, *, _name: str = target_name, **kw):
         for op, target_name in cls.SIMPLE_OPS.items():
             setattr(cls, methodname(op), make_impl(op, target_name))
 
+        # Define binary op methods, only if BINARY_INFIX_OPS is set on the
+        # compiler class.
+        if binops := cls.__dict__.get("BINARY_INFIX_OPS", {}):
```

Comment:
Is this checking specifically for subclass overrides of this attribute?

Reply:
Yeah. If we did `cls.BINARY_INFIX_OPS` it would define the methods in every subclass (since no subclasses currently define `BINARY_INFIX_OPS`), clobbering any explicit `visit_*` methods the subclass defined. We'd need to do this for `SIMPLE_OPS` etc... too, except that every compiler subclass currently sets that attribute themselves with at least one override.